backends/rabbitmq/read.go

```diff
@@ -34,38 +35,26 @@ func Read(opts *cli.Options) error {
 		}
 	}
 
-	ch, err := connect(opts)
-	if err != nil {
-		return errors.Wrap(err, "unable to complete initial connect")
-	}
+	r, err := New(opts, md)
 
-	r := &RabbitMQ{
-		Options: opts,
-		Channel: ch,
-		MsgDesc: md,
-		log:     logrus.WithField("pkg", "rabbitmq/read.go"),
+	if err != nil {
+		return err
```

Comment:
Let's wrap this in `errors.Wrap(err, "informative message about where this occurred")`